libbeat/processors/add_kubernetes_metadata/podwatcher.go

```diff
@@ -189,18 +197,65 @@ func (p *PodWatcher) worker() {
 	}
 }
 
+// Check annotations flagged as deleted for their last access time, fully delete
+// the ones older than p.cleanupTimeout
+func (p *PodWatcher) cleanupWorker() {
+	for {
+		// Wait a full period
+		time.Sleep(p.cleanupTimeout)
+
+		select {
+		case <-p.ctx.Done():
+			return
+		default:
+			// Check entries for timeout
+			var toDelete []string
+			timeout := time.Now().Add(-p.cleanupTimeout)
+			p.annotationCache.RLock()
```

Comment:
Not for this PR, but the code looks so similar to the other cleanupWorker, I'm thinking if parts could be shared.